index.js

```diff
@@ -33,6 +33,14 @@ class Topic extends EventEmitter {
     this._startDht()
     if (!this.announce || opts.lookup) this._startMdns()
     if (this.announce) process.nextTick(this._fireAnnounce.bind(this))
+
+    this._onannounce = (target, peer) => {
+      if (target.equals(this.key)) {
```

Comment:
Not sure if we should go for more optimal check here, or stay with the event listener that is very easy to understand.